src/mpid/ch4/shm/ipc/xpmem/xpmem_init.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "xpmem_noinline.h"
+#include "mpidu_init_shm.h"
+#include "xpmem_seg.h"
+#include "shm_control.h"
+
+int MPIDI_IPC_xpmem_mpi_init_hook(int rank, int size, int *tag_bits)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int i;
+    bool anyfail = false;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPC_XPMEM_MPI_INIT_HOOK);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPC_XPMEM_MPI_INIT_HOOK);
+    MPIR_CHKPMEM_DECL(3);
+
+#ifdef MPL_USE_DBG_LOGGING
+    MPIDI_CH4_SHM_IPC_XPMEM_GENERAL = MPL_dbg_class_alloc("SHM_IPC_XPMEM", "shm_ipc_xpmem");
+#endif /* MPL_USE_DBG_LOGGING */
+
+    /* Try to share entire address space */
+    MPIDI_IPC_xpmem_global.segid = xpmem_make(0, XPMEM_MAXADDR_SIZE, XPMEM_PERMIT_MODE,
+                                              MPIDI_IPC_XPMEM_PERMIT_VALUE);
+    /* 64-bit segment ID or failure(-1) */
+    MPIR_ERR_CHKANDJUMP(MPIDI_IPC_xpmem_global.segid == -1, mpi_errno, MPI_ERR_OTHER,
+                        "**xpmem_make");
+    XPMEM_TRACE("init: make segid: 0x%lx\n", (uint64_t) MPIDI_IPC_xpmem_global.segid);
+
+    MPIDU_Init_shm_put(&MPIDI_IPC_xpmem_global.segid, sizeof(xpmem_segid_t));
```

Comment:
Why is this in the `MPIDI_IPC_xpmem_` namespace?

Reply:
I am not sure but segid is only used by xpmem. We can probably put it in the IPC global structure but only xpmem will use it. Do you want to do that?

Reply:
Umm.  No, my comment was why is this namespace `MPIDI_IPC_xpmem_` for something that's not used by IPC?  It's internal to XPMEM.

Reply:
The namespace is not for what is used by IPC, is for all submodule functions and data structures. Do you want to make the explicit distinction between what is submodule private and what is IPC layer accessible?

Reply:
Of course.  :-)   Isn't that just good software engineering?  XPMEM private data structures and functions should not be visible to the IPC layer.

Reply:
In my opinion, removing the prefix `MPIDI_IPC_` can be risky because it may conflict with another inlined function with the same name. We don't know what functions are defined in UCX for XPMEM,  for example. Although such a case should rarely happen. 

If we remove the namespace only for noninline internal functions, it does not look consistent.

Reply:
We may use another namespace such as `MPIDI_IPC_ xpmemi_` to indicate it is ipc/xpmem internal, if it helps.

Reply:
I never said "remove the namespace".  I asked why it is in the `MPIDI_IPC_xpmem_` namespace.  Would you name an MPICH internal function as `MPI_`?  Please also note my other comment about why these are inline functions.

If this is causing so much confusion, I'd really recommend offline discussion.  Discussing over GitHub is not efficient.